com.sap.cloud.lm.sl.cf.persistence/src/main/java/com/sap/cloud/lm/sl/cf/persistence/jclouds/providers/aliyun/blobstore/AliOSSBlobStore.java

```diff
@@ -0,0 +1,328 @@
+                                                                     .stream()
+                                                                     .map(ossObjectSummary -> toStorageMetadata(oss, container, ossObjectSummary))
+                                                                     .collect(Collectors.toList());
+            return new PageSetImpl<>(storageMetadataList, objectListing.getNextMarker());
+        });
+    }
+
+    @Override
+    public boolean blobExists(String container, String name) {
+        return doOssOperation(oss -> oss.doesObjectExist(container, name));
+    }
+
+    @Override
+    public String putBlob(String container, Blob blob) {
+        return doOssOperation(oss -> {
+            try {
+                ObjectMetadata objectMetadata = createObjectMetadataFromBlob(blob);
+                PutObjectRequest request = new PutObjectRequest(container, blob.getMetadata()
+                                                                               .getProviderId(), blob.getPayload()
+                                                                                                     .openStream(), objectMetadata);
+                PutObjectResult result = oss.putObject(request);
+                return result.getETag();
+            } catch (IOException e) {
+                throw new IllegalStateException(e);
+            }
+        });
+    }
+
+    @Override
+    public String putBlob(String container, Blob blob, PutOptions options) {
+        return putBlob(container, blob);
+    }
+
+    @Override
+    public Blob getBlob(String container, String name, GetOptions options) {
+        return doOssOperation(oss -> {
+            GetObjectRequest req = toGetObjectRequest(container, name, options);
+            OSSObject object = oss.getObject(req);
+            return convertToBlob(object);
+        }, false);
+
+    }
+
+    @Override
+    public void removeBlob(String container, String name) {
+        doOssOperation(oss -> {
+            oss.deleteObject(container, name);
+            return null;
+        });
+    }
+
+    private <R> R doOssOperation(Function<OSS, R> function) {
+        return doOssOperation(function, true);
+    }
+
+    private <R> R doOssOperation(Function<OSS, R> function, boolean shutdownClient) {
+        OSS ossClient = aliOSSApi.getOSSClient();
+        R result = function.apply(ossClient);
+        if (shutdownClient) {
+            ossClient.shutdown();
+        }
+        return result;
+    }
+
+    private GetObjectRequest toGetObjectRequest(String container, String name, GetOptions options) {
+        GetObjectRequest request = new GetObjectRequest(container, name);
+        if (options.getIfModifiedSince() != null) {
+            request.setModifiedSinceConstraint(options.getIfModifiedSince());
+        }
+        if (!options.getRanges()
+                    .isEmpty()) {
+            String[] ranges = options.getRanges().get(0).split("-");
```

Comment:
Why is always the first element used? get(0)

Reply:
Because it contains only one element